### DISCUSSION TITLE
Most Used Languages works wrong

### DISCUSSION THREAD

**Discussion by berkaygurcan** [anuraghazra/github-readme-stats#1848] [Q&A] [answered]
My Java projects show python even though there are more. I don't think it's related to Gitlab. How can I solve this problem?
`https://github-readme-stats.vercel.app/api/top-langs?username=berkaygurcan&show_icons=true&locale=en&layout=compact`

**Comment by codereptile** [comment]
I think that is because in [this repo](https://github.com/berkaygurcan/my-first-blog) you have a `myvenv` virtualenv that has tonns of python code in it. Since this project calculates the percentage based on lines of code, this venv drastically shifts your most used languages to python.

**Comment by berkaygurcan** [comment]
Thank you. What do you suggest I do, does it make any difference if I make the repo private?

**Comment by codereptile** [comment]
In general, I would add vevn to `.gitignore` (not only for this project, but for all future projects) and list all dependencies in a `requirements.txt`. This would help reduce repo size, make language statistics more accurate, but you will still be able to reproduce the same vevn any time you need from `requrements.txt`. Also, it will be easier to manage your dependencies in case they change, as all you will need to do is add or remove a line, instead of changing the whole venv.

This is a general advice and you probably should start doing so independently of whether the stats are rendered correctly.
However, if for some reason after you do so, the stats are still incorrect, yes, by default the app doesn't count private repos, so making it private could be a solution.

**Comment by AbhishekPowar** [comment]
@berkaygurcan  

You can try this. 
https://github-readme-stats.vercel.app/api/top-langs?username=berkaygurcan&show_icons=true&locale=en&layout=compact&hide=python

Note : This hides all python contributions from stat

[![GitHub stats](https://github-readme-stats.vercel.app/api/top-langs?username=berkaygurcan&show_icons=true&locale=en&layout=compact&hide=python)](https://github.com/berkaygurcan/github-readme-stats)

**Comment by MolotovCherry** [comment, marked as answer]
Hi, github uses Linguist. They have provided an option to hide paths from being checked in the stats. You can use this in your offending repo to fix the stats. It should trickle its way back up

https://github.com/github/linguist/blob/master/docs/overrides.md

`linguist-vendored` is the one that will hide it from stats
